sw_parser/views.py

```diff
@@ -1274,6 +1288,27 @@ def view_elemental_rift_log(request, rift_slug, mine=False):
             grade_item_table[monster_drop['log__grade']]['items'][monster_drop['monster__com2us_id']]['avg_drop'] = chance_to_drop
             grade_item_table[monster_drop['log__grade']]['items'][monster_drop['monster__com2us_id']]['chance_drop'] = chance_to_drop * 100
 
+        for rune_drop in RiftDungeonRuneDrop.objects.filter(log__in=runs).values('log__grade').annotate(
+            avg_stars=Avg('stars'),
+            avg_quality=Avg('quality')
+        ):
+            # at a glance, what grade/rift you need to start farming quality runes.
+            grade_rune_table[rune_drop['log__grade']]['avg_stars'] = rune_drop['avg_stars']
+            grade_rune_table[rune_drop['log__grade']]['avg_quality'] = rune_drop['avg_quality']
+            # 2.6 is dropping more Heros than Rares, but at 2.5 the Rares are more noticable
+            grade_rune_table[rune_drop['log__grade']]['avg_rarity'] = RuneDrop.QUALITY_CHOICES[int(rune_drop['avg_quality']+0.4)][1]
+            
+        for rune_drop in RiftDungeonRuneDrop.objects.filter(log__in=runs).values('type', 'log__grade').annotate(
+            avg_quality=Avg('quality'),
+            occurences=Count('pk')
+        ):
+            # Just in case the average quality is different per set, show the rune-bg behind the icon.
+            rune_list[rune_drop['type']]['avg_quality'] = RuneDrop.QUALITY_CHOICES[int(rune_drop['avg_quality']+0.4)][1]
```

Comment:
The rune_list is initially populated from all Elemental Rifts, but this populates avg_quality on a filtered set of Elemental Rifts. If a certain rune did not drop in the filtered set, it will not get an avg_quality assigned which leads to a bad URL for the background image. I can see a couple ways to fix this:

1. Default to an avg_quality of 0 and therefore Normal background when you initially create rune_list
2. Check in the template if avg_quality exists and if not, use Normal background.
3. Only populate rune_list with runes that are present in the filtered runs. This is my least favorite option - I like having the list have the same columns across all reports. 

This situation might be pretty unlikely when viewing the global log reports, but it could definitely happen when viewing personal logs where there may be very few logs available.

Reply:
I thought something was fishy when I checked my local runs for a different rift.  

Another thing I was messing around with was an "All Runs" row above "SSS" which would display the chance per run of each rune type.  
How about it?

Reply:
That sounds good!